jiant/tasks/tasks.py

```diff
@@ -293,6 +295,41 @@ def handle_preds(self, preds, batch):
         """
         return preds
 
+    def set_instance_generator(
+        self, dataset_name: str, instance_generator: Iterable, phase: str = None
+    ):
+        """Takes a data instance generator and stores it in a private field of this Task instance
+
+        Parameters
+        ----------
+        dataset_name : string
```

Comment:
I'm more familiar with 'dataset' referring to a full collection, not a specific split subset of the data. How about something like 'split_name' instead?

Reply:
Otherwise, it's plausible that future code editors would try setting something like dataset_name = "mnli".

Reply:
👍, done w/ c5d6822.